contrib/config/test/kv_store_xds_delegate_integration_test.cc

```diff
@@ -0,0 +1,340 @@
+      EXPECT_TRUE(compareSotwDiscoveryRequest(
+          /*expected_type_url=*/Config::TypeUrl::get().Runtime,
+          /*expected_version=*/"",
+          /*expected_resource_names=*/{"some_rtds_layer"}, /*expect_node=*/true,
+          /*expected_error_code=*/Grpc::Status::WellKnownGrpcStatus::Ok,
+          /*expected_error_message=*/"", rtds_stream_.get()));
+      auto rtds_resource = TestUtility::parseYaml<envoy::service::runtime::v3::Runtime>(R"EOF(
+          name: some_rtds_layer
+          layer:
+            foo: bar
+            baz: meh
+      )EOF");
+      sendSotwDiscoveryResponse<envoy::service::runtime::v3::Runtime>(
+          Config::TypeUrl::get().Runtime, {rtds_resource}, "1", rtds_stream_.get());
+    }
+  };
+
+  initialize();
+
+  // Wait until the discovery responses have been processed.
+  test_server_->waitForCounterGe(
+      "cluster.cluster_0.client_ssl_socket_factory.ssl_context_update_by_sds", 1);
+  test_server_->waitForCounterGe("runtime.load_success", 2);
+
+  // Verify that the xDS resources are used by Envoy.
+  checkSecretExists(std::string(CLIENT_CERT_NAME), /*version_info=*/"1");
+  EXPECT_EQ("bar", getRuntimeKey("foo"));
+  EXPECT_EQ("yar", getRuntimeKey("bar"));
+  EXPECT_EQ("meh", getRuntimeKey("baz"));
+
+  // Send an update to the RTDS resource, from the RTDS cluster to the Envoy test server.
+  EXPECT_TRUE(compareSotwDiscoveryRequest(
+      /*expected_type_url=*/Config::TypeUrl::get().Runtime, /*expected_version=*/"1",
+      /*expected_resource_names=*/{"some_rtds_layer"}, /*expect_node=*/false,
+      /*expected_error_code=*/Grpc::Status::WellKnownGrpcStatus::Ok,
+      /*expected_error_message=*/"", rtds_stream_.get()));
+  auto rtds_resource = TestUtility::parseYaml<envoy::service::runtime::v3::Runtime>(R"EOF(
+    name: some_rtds_layer
+    layer:
+      baz: saz
+  )EOF");
+  sendSotwDiscoveryResponse<envoy::service::runtime::v3::Runtime>(
+      Config::TypeUrl::get().Runtime, {rtds_resource}, "2", rtds_stream_.get());
+  test_server_->waitForCounterGe("runtime.load_success", 3);
+
+  EXPECT_EQ("whatevs", getRuntimeKey("foo"));
+  EXPECT_EQ("yar", getRuntimeKey("bar"));
+  EXPECT_EQ("saz", getRuntimeKey("baz"));
+
+  // Kill the current test server, and restart it using the same configuration.
+  shutdownAndRestartTestServer();
+
+  // Wait until SDS and RTDS have been loaded from the KV store and updated the Envoy instance.
+  test_server_->waitForCounterGe(
+      "cluster.cluster_0.client_ssl_socket_factory.ssl_context_update_by_sds", 1);
+  // Two runtime loads are expected, one for the admin layer and one for the RTDS layer.
+  test_server_->waitForCounterGe("runtime.load_success", 2);
+
+  // Verify that the latest resource values are used by Envoy.
+  checkSecretExists(std::string(CLIENT_CERT_NAME), /*version_info=*/"1");
+  EXPECT_EQ("whatevs", getRuntimeKey("foo"));
+  EXPECT_EQ("yar", getRuntimeKey("bar"));
+  EXPECT_EQ("saz", getRuntimeKey("baz"));
```

Comment:
I guess this is the indication that the data was loaded from the KV-store.
To enhance observability I suggest adding a counter whenever data is loaded from the KV-store (for each type).

Reply:
I added some stats